llvm/include/llvm/Analysis/InstSimplifyFolder.h

```diff
@@ -117,6 +117,11 @@ class InstSimplifyFolder final : public IRBuilderFolder {
     return simplifyCastInst(Op, V, DestTy, SQ);
   }
 
+  Value *FoldBinaryIntrinsic(Intrinsic::ID ID, Value *LHS,
+                             Value *RHS) const override {
+    return simplifyBinaryIntrinsic(ID, LHS->getType(), LHS, RHS, SQ, nullptr);
```

Comment:
`LHS->getType()` is incorrect for `@llvm.is.fpclass`. I don't know why `abs/ctlz/cttz/is.fpclass` are treated as binary intrinsics.

Reply:
I've updated the `FoldBinaryIntrinsic` definition to include a `Type *` parameter, which is be passed from `CreateBinaryIntrinsic` using `F->getReturnType()`.